examples/taxi-ride-akka/README.md

```diff
@@ -0,0 +1,95 @@
+## Akka implementation of taxi ride application
+
+This example demonstrates how `taxi-ride` application initially implemented in flink can be implemented using Akka.
+
+The implementation combines two streams `TaxiRide` and `TaxiFare` and for each rideID calculates
+a fare for a given ride 
+
+In the case of Flink we used `RichCoFlatMapFunction` to combine the two streams, here
+we are using `RideShare` actor for the same purpose. If we compare implementation of the actor 
+with the implementation of Flink's `RichCoFlatMapFunction`, we will see that they are very similar.
+The missing part here is persistence. While Flink implementation provides automatic snapshotting, Akka
+requires implementation of an explicit persistence support, which is not currently supported
+by Cloudflow (but can be easily added manually).
+
+Main flow of Flink implementation is very elegant and rasy to read   
+````
+      val rides: DataStream[TaxiRide] =
+        readStream(inTaxiRide)
+          .filter { ride ⇒
+            ride.isStart.booleanValue
+          }
+          .keyBy("rideId")
+
+      val fares: DataStream[TaxiFare] =
+        readStream(inTaxiFare)
+          .keyBy("rideId")
+
+      val processed: DataStream[TaxiRideFare] =
+        rides
+          .connect(fares)
+          .flatMap(new EnrichmentFunction)
+
+      writeStream(out, processed)
+````
+
+Akka implementation provides the same functionality, with the following code:
+````
+      val rides = shardedSourceWithCommittableContext(inTaxiRide, entity).via(ridesFlow)
+
+      val fares = shardedSourceWithCommittableContext(inTaxiFare, entity).via(faresFlow)
+
+      Merger.source(Seq(rides, fares)).to(committableSink(out))
+````
+Note that here there is no explicit `keyBy` (we will discuss it later). 
+Unlike Flink implementation, where streams are connected before invoking, in the case of
+Akka, we process each stream independantly and then merge results afterwards.
+Finally individual flow looks like following:
+
+````
+      FlowWithCommittableContext[TaxiRide]
+        .mapAsync(1)(msg ⇒ {
+          val actor = sharding.entityRefFor(typeKey, msg.rideId.toString)
+          actor.ask[Option[TaxiRideFare]](ref => ProcessRide(ref, msg))
+        }).collect{ case Some(v) => v }
+````
+Here, `keyBy` is implemented through actor selection - we are using`rideId` is the actor ID.
+So the implementation provides the same functionality (except for automatic persistence, 
+which can be easily added), but with less user friendly syntax. Flink's implementation hides
+all of the sharding details by usage of `keyBy`.
+On another hand, FLink uses static topology for applications, which means that scaling requires
+application restarts, while in the case of Akka, adding additional instance can be done without 
```

Comment:
What do you mean here, that existing, running actor systems as part of the cluster don't have to restart?

Reply:
exactly. You can add an additional instance without any restarts of the running ones